integration-tests/test_collection.py

```diff
@@ -152,8 +159,16 @@ def test_branch_info(insights_client):
     branch_info_path = os.path.join(directory_name, "branch_info")
     with open(branch_info_path, "r") as file:
         data = json.load(file)
-        assert data["remote_branch"] == -1, "Incorrect remote_branch value"
-        assert data["remote_leaf"] == -1, "Incorrect remote_leaf value"
+        if "satellite" in test_config.environment:
+            assert data["product"]["type"] == "Satellite"
+            assert isinstance(uuid.UUID(data["remote_branch"]), uuid.UUID)
+            cmd = "subscription-manager identity"
+            output = subprocess.check_output(cmd, universal_newlines=True, shell=True)
+            system_identity = output.split("\n")[0].split(":")[1].strip()
+            assert data["remote_leaf"] == system_identity
```

Comment:
I just added a `SubscriptionManager.uuid` property in pytest-client-tools, which returns a `uuid.UUID` object for a registered system; now you can simply use it as:
```python
def test_branch_info(insights_client, test_config, subman):
   ...
            assert data["remote_leaf"] == subman.uuid
```